Client/src/Pages/PageSpeed/index.jsx

```diff
@@ -0,0 +1,153 @@
+import { Box, Grid, Stack, Typography } from "@mui/material";
+import Fallback from "../../Components/Fallback";
+import { useTheme } from "@emotion/react";
+import PageSpeedIcon from "../../assets/icons/page-speed.svg?react";
+
+import "./index.css";
+import { formatDate, formatDurationRounded } from "../../Utils/timeUtils";
+import { StatusLabel } from "../../Components/Label";
+
+const Card = ({ data }) => {
+  const theme = useTheme();
+
+  /**
+   * Helper function to get duration since last check or the last date checked
+   * @param {Array} checks Array of check objects.
+   * @param {boolean} duration Whether the function should return the duration since last checked or the date itself
+   * @returns {number} Timestamp of the most recent check.
+   */
+  const getLastChecked = (checks, duration = true) => {
+    if (!checks || checks.length === 0) {
+      return 0; // Handle case when no checks are available
+    }
+
+    // Data is sorted oldest -> newest, so last check is the most recent
+    if (!duration) {
+      return new Date(checks[checks.length - 1].createdAt);
+    }
+    return new Date() - new Date(checks[checks.length - 1].createdAt);
+  };
+
+  return (
+    <Grid item lg={6} flexGrow={1}>
+      <Stack direction="row" gap={theme.gap.medium} p={theme.gap.ml}>
+        <PageSpeedIcon style={{ width: theme.gap.ml, height: theme.gap.ml }} />
+        <Box flex={1}>
+          <Stack direction="row" justifyContent="space-between">
+            <Typography component="h2" mb={theme.gap.xs}>
+              {data.name}
+            </Typography>
+            {/* TODO - fix prop-type error */}
```

Comment:
We could just remove the acceptable `strings` from the label component, I don't think there's any real good reason to limit what strings we can use there :thinking:

Reply:
Yea, that's probably the move. I'm thinking we should define some styles types in the component itself as well, that way there's no need to pass a color variable for background/dot at all.